client/src/state/schema/SchemaReducer.js

```diff
@@ -12,6 +12,18 @@ export default function schemaReducer(state = initialState, action = null) {
       }));
     }
 
+    case ACTION_TYPES.CLEAR_SCHEMA: {
+      return deepFreeze(Object.entries(state).reduce((prev, [key, value]) => {
```

Comment:
Requires quite a bit of thinking. Can you help break this down a bit and/or a unit test would help as well.